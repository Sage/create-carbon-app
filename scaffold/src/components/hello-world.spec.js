import React from 'react';
import { shallow } from 'enzyme';
import Button from 'carbon-react/lib/components/button';
import HelloWorld from './hello-world';

// Demonstration of a simple test for the HelloWorld component:
describe('HelloWorld', () => {
  const renderShallow = () => (
    shallow(<HelloWorld />)
  );

  it('renders a title and button', () => {
    const wrapper = renderShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('says hello on click of the button', () => {
    global.alert = jest.fn();
    const wrapper = renderShallow();
    wrapper.find(Button).simulate('click');
    expect(global.alert).toHaveBeenCalledWith('Hello world!');
  });
});
