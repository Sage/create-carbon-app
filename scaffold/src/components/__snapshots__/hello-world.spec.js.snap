// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld renders a title and button 1`] = `
<AppWrapper>
  <h1>
    Hello world!
  </h1>
  <Button
    as="secondary"
    disabled={false}
    onClick={[Function]}
    size="medium"
    subtext=""
    theme="blue"
  >
    Click me!
  </Button>
</AppWrapper>
`;
